### PR TITLE
fix: Add Spinner

### LINKED ISSUE #178: Add Spinner after "Add X Files" has been clicked from Picker
After clicking that button, there is a delay that occurs (to load). Add a spinner there and hide the button to indicate that the component is loading.
image

### PR DESCRIPTION
**Pull request by maheshchandra10**
Fixes #178

### CODE REVIEW

**Comment by maheshchandra10** [comment]
The button should look like this after clicking it-

image

in #178 , you have asked to remove the button and then add a spinner, but then the `Cancel` button which is next to the `Add X Files` button will be **shifted to the left side**.

The button with the loader is somewhat of the same size as the `Add X Files` button, which will not shift the `Cancel` button, maintaining the responsiveness of the buttons.

If you want to fix the `Cancel` button to one place and then want to show only the loader (without the button), then the **responsiveness** of these two buttons will be compromised I believe.

File: src/components/UpupUploader/FileBrowser/index.tsx [hunk 1]
```diff
@@ -144,12 +147,22 @@ const FileBrowser = ({
...
+                                <path d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z" fill="currentColor"></path>
+                            </svg>
+                            Loading...
+                        </button>)}
```
Comment (bassem97) [inline src/components/UpupUploader/FileBrowser/index.tsx]: wrap this into a component for better code readability

**Review by bassem97 — CHANGES_REQUESTED** [review]
Great job @maheshchandra10

**Comment by bassem97** [comment]
> The button should look like this after clicking it-
> 
> image
> 
> in #178 , you have asked to remove the button and then add a spinner, but then the `Cancel` button which is next to the `Add X Files` button will be **shifted to the left side**.
> 
> The button with the loader is somewhat of the same size as the `Add X Files` button, which will not shift the `Cancel` button, maintaining the responsiveness of the buttons.
> 
> If you want to fix the `Cancel` button to one place and then want to show only the loader (without the button), then the **responsiveness** of these two buttons will be compromised I believe.

i suggest to make the Add X file button the same width as the loading button so it doesn't shift .

**Comment by maheshchandra10** [comment]
@bassem97 Separated the spinner to a different component.

I have also made the buttons width to be equal using `w-32` _(I used w-32 coz it was looking best for both the buttons, one with the `Add X Files` text and one with the `spinner`)_.

**Review by MohamedBechirMejri — APPROVED** [review]

**Review by bassem97 — APPROVED** [review]
Great job again !

**Review by bassem97 — APPROVED** [review]

**Pull request merged**